### PR TITLE
Ppp and dg shawn

### PR DESCRIPTION
**Pull request by lwlshawn**
Rewrote my part of the DG and shifted the sections as discussed in the call this morning.

### CODE REVIEW

**Comment by lwlshawn** [comment]
image

Image to help view what the sequence and activity diagram look like.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -185,75 +185,104 @@ How the Storage component works:
...
+
+We updated the equality check to account for this, and also updated the input validation for `Name` to allow at most one
+space between blocks of characters (previously `John   Doe` would be different from `John Doe`. We felt this likely to be
+a mistake and should be avoided).
+
+### Add Order feature
+As mentioned above, to add an `Order`, the `Person` the `Order` is addressed to should already be in `SalesNote`. The
+following is a sequence diagram showing the execution of the command:
+
+![AddOrderSequenceDiagram](images/AddOrderSequenceDiagram.png)
```
Comment (ngchisern) [inline docs/DeveloperGuide.md:267]: I just realised if you want to include the `hasorder` and `haspersonwithname` methods in the sequence diagram, you have to wrap the `alt` frames around each of them as well.
Screenshot 2021-11-08 at 11 03 54 AM

**Review by ngchisern — CHANGES_REQUESTED** [review]
Suggested some changes below. We might also want to decide whether we want to use 'an `Order`' or 'a `Order`' in the DG for consistency (The current DG uses 'an `Order`' now).

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (lwlshawn) [reply docs/DeveloperGuide.md:267]: 'an Order' is more correct I believe, but i'm not sure if alt is the correct type of annotation to add here

**Review by ngchisern — APPROVED** [review]
LGTM. I'll fix the 'a`order`' issues when I'm formatting the DG later.

**Pull request merged**